business_rules/operators.py

```diff
@@ -938,6 +939,68 @@ def is_ordered_by(self, other_value: dict) -> pd.Series:
     def is_not_ordered_by(self, other_value: dict) -> pd.Series:
         return ~self.is_ordered_by(other_value)
 
+    @type_operator(FIELD_DATAFRAME)
+    def target_is_sorted_by(self, other_value: dict) -> pd.Series:
+        """
+         Checking if  SESEQ is  sorted based on SESTDTC within USUBJID
+        """
+
+        target: str = self.replace_prefix(other_value.get("target"))
+        columns = list(other_value["user_defined_columns"].keys())
+
+        for index in range(0, len(columns), 2):
+            within: str = self.replace_prefix(columns[index])
+            comparator: str = self.replace_prefix(columns[index+1])
+
+            temp_seseq = self.create_count_map(within)
+
+            if type(self.value[within][0]) is str:
+                if self.value[within][0].isupper():
+                    for i in range(len(temp_seseq)):
+                        temp_seseq[i] = chr(ord("@")+temp_seseq[i])
+                if self.value[within][0].islower():
+                    for i in range(len(temp_seseq)):
+                        temp_seseq[i] = chr(ord("`")+temp_seseq[i])
+
+            sorted_df = self.value.sort_values(by=[within], ascending=True,na_position='last')
+            grouped_df = sorted_df.sort_values(by=[comparator], ascending=True).groupby([within])
+            temp_target = self.create_temp_target(within, comparator, temp_seseq, grouped_df)
```

Comment:
Shouldn't the ascending and na_position come from the definition in the user_defined_columns/comparator?